frontend/src/component/personalDashboard/FlagMetricsChart.tsx

```diff
@@ -48,73 +57,144 @@ const data = {
     ],
 };
 
-const createBarChartOptions = (theme: Theme): ChartOptions<'bar'> => ({
-    plugins: {
-        legend: {
-            position: 'bottom',
-            labels: {
-                color: theme.palette.text.primary,
-                pointStyle: 'circle',
-                usePointStyle: true,
-                boxHeight: 6,
-                padding: 15,
-                boxPadding: 5,
-            },
-        },
-        tooltip: {
-            enabled: false,
-        },
-    },
-    responsive: true,
-    scales: {
-        x: {
-            stacked: true,
-            ticks: {
-                color: theme.palette.text.secondary,
-            },
-            grid: {
-                display: false,
-            },
-        },
-        y: {
-            stacked: true,
-            ticks: {
-                color: theme.palette.text.secondary,
-                maxTicksLimit: 5,
-                callback: formatTickValue,
-            },
-            grid: {
-                drawBorder: false,
-            },
-        },
-    },
-    elements: {
-        bar: {
-            borderRadius: 5,
-        },
-    },
-    interaction: {
-        mode: 'index',
-        intersect: false,
-    },
-});
-
 export const PlaceholderFlagMetricsChart = () => {
     const theme = useTheme();
 
     const options = useMemo(() => {
-        return createBarChartOptions(theme);
+        return createPlaceholderBarChartOptions(theme);
     }, [theme]);
 
     return (
-        <Bar
-            data={data}
-            options={options}
-            aria-label='A bar chart with a single feature flag exposure metrics'
+        <>
+            <Typography sx={{ mb: 4 }}>Feature flag metrics</Typography>
+            <Bar
+                data={placeholderData}
+                options={options}
+                aria-label='A placeholder bar chart with a single feature flag exposure metrics'
+            />
+        </>
+    );
+};
+
+const useMetricsEnvironments = (project: string) => {
+    const [environment, setEnvironment] = useState<string | null>(null);
+    const { environments } = useProjectEnvironments(project);
+    const activeEnvironments = environments.filter((env) => env.enabled);
+    const firstProductionEnvironment = activeEnvironments.find(
+        (env) => env.type === 'production',
+    );
+
+    useEffect(() => {
+        if (firstProductionEnvironment) {
+            setEnvironment(firstProductionEnvironment.name);
+        } else if (activeEnvironments.length > 0) {
+            setEnvironment(activeEnvironments[0].name);
+        }
+    }, [JSON.stringify(activeEnvironments)]);
+
+    return { environment, setEnvironment, activeEnvironments };
+};
+
+const useFlagMetrics = (
```

Comment:
copied all this code from the feature metrics. One difference is that now we have a hook that prepares chart options + chart data in one function instead of going through multiple components